### ISSUE TITLE
idea: A Cargo plugin for generating a schema file

### ISSUE THREAD

**Issue by ranger-ross** [juhaku/utoipa#1258]
Hey there :wave: 

I am using utoipa for generating openapi schema during CI to be uploaded to a "build metadata" database.

Currently, I have something like 
```rs
// Semi pseudo code 
fn main() {
     let openapi = ApiDoc::openapi();

     if std::env::args().contains("--dump-openapi-schema") {
           write_file("target/openapi.json", openapi);
           return;
     }

    // The API server code
}
```

This works but we have a lot of services owned by many teams so having this code pasted across all of our repos is a bit burdensome for maintenance. I was debating creating an internal crate for this to help standardize it internally. 

But I had an idea pop into my mind for creating a utoipa cargo plugin that could have some common commands like generating a schema.
```sh
cargo utoipa generate
# Generates the openapi file in $CARGO_TARGET_DIR/openapi.json
```

I haven't given it a ton of thought on what the actually implementation would look like, but I figured I would post the idea to the issue tracker incase other utoipa users would find it useful. :thinking:

**Comment by juhaku** [comment]
Good idea, I have been pondering on same matter once in a while, but never got into it, as it seems that there has been other priorities taking before it. Sure I am open for PR:s or other third-party contributions for making such plugin possible.
